superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -124,6 +208,187 @@ export default class ResultSet extends React.PureComponent<
     }
   }
 
+  handleOnChangeAutoComplete = () => {
+    this.setState({ datasetToOverwrite: {} });
+  };
+
+  handleOverwriteDataset = async () => {
+    const { sql, results } = this.props.query;
+    const { datasetToOverwrite } = this.state;
+
+    if (
+      Object.keys(datasetToOverwrite).length === 0 &&
+      datasetToOverwrite.constructor === Object
```

Comment:
will `Object.keys(datasetToOverwrite).length === 0` ever be true and `datasetToOverwrite.constructor === Object` be false? Seems like if the former is true, then the constructor will always be an Object?

Reply:
Good catch i actually figured out how to prevent this via onChangeAutocomplete. Users have to choose a value from the dropdown before progressing to the overwrite page. So we don't need this condition.